olive/engine/engine.py

```diff
@@ -422,9 +429,6 @@ def run_no_search(
                     json.dump(signal.to_json(), f, indent=4)
                 logger.info("Saved evaluation results of output model to %s", results_path)
 
-            self.cache.save_model(model_id=model_ids[-1], output_dir=flow_output_dir, overwrite=True)
```

Comment:
is this is not present, then I don't think this `pass_flow` nesting is present? 
![image](https://github.com/user-attachments/assets/1ca0bf83-2d3e-49b7-a906-14d4a83aafa2)

not sure if we should keep it or not